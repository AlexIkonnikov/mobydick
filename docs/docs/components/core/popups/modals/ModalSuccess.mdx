import ModalSuccess from '../assets/ModalSuccess.png';

<img src={ModalSuccess} width="50%"/>

Reusable ModalSuccess component

## Example

```tsx
import {
  Button,
  createStyles,
  ModalSuccess,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const ModalsWidget = () => {
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Button
        text={'ModalSuccess'}
        onPress={() =>
          openPopup({
            Content: props => (
              <ModalSuccess
                title={'ModalSuccess'}
                descriptionText={'descriptionText'}
                {...props}></ModalSuccess>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces}) => ({container: {gap: spaces.Space8}}));

```

## `Props`

### [`IContentProps`](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/popups/types.ts#L6)

Inherits [IContentProps](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/popups/types.ts#L11)

### <div class="label required basic">Required</div>**`title`**

| TYPE   |
|:-------|
| string |

Title in ModalSuccess

### <div class="label required basic">Required</div>**`descriptionText`**

| TYPE   |
|:-------|
| string |

DescriptionText in ModalSuccess

### `buttonText`

| TYPE   | DEFAULT |
|:-------|:--------|
| string | OK      |

Text for button
