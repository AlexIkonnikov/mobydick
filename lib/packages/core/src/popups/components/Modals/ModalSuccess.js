"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ModalBase_1 = require("../ModalBase");
const useStyles_1 = __importDefault(require("../../../styles/hooks/useStyles"));
const stylesCreate_1 = __importDefault(require("./stylesCreate"));
const ModalSuccess = props => {
    const { onClose, title, descriptionText, buttonText } = props;
    const [styles] = (0, useStyles_1.default)(stylesCreate_1.default);
    return (<ModalBase_1.ModalBase overlayStyle={styles.overlayStyle} containerStyle={styles.container} {...props}>
      <ModalBase_1.ModalBase.AlertContent />

      <ModalBase_1.ModalBase.TextContent title={title} descriptionText={descriptionText}/>

      <ModalBase_1.ModalBase.VerticalButtonsView>
        <ModalBase_1.ModalBase.VerticalButton onPress={onClose} text={buttonText || 'OK'}/>
      </ModalBase_1.ModalBase.VerticalButtonsView>
    </ModalBase_1.ModalBase>);
};
exports.default = ModalSuccess;
