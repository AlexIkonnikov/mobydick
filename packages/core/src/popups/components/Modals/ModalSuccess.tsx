import React, {FC} from 'react';

import {ModalBase} from '../ModalBase';
import {IContentProps} from '../../types';
import useStyles from '../../../styles/theme/hooks/useStyles';

import stylesCreate from './stylesCreate';
import {IModalProps} from './types';

const ModalSuccess: FC<IContentProps & IModalProps> = props => {
  const {onClose, title, descriptionText, buttonText} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <ModalBase
      overlayStyle={styles.overlayStyle}
      containerStyle={styles.container}
      {...props}>
      <ModalBase.AlertContent />

      <ModalBase.TextContent title={title} descriptionText={descriptionText} />

      <ModalBase.VerticalButtonsView>
        <ModalBase.VerticalButton onPress={onClose} text={buttonText || 'OK'} />
      </ModalBase.VerticalButtonsView>
    </ModalBase>
  );
};

export default ModalSuccess;
