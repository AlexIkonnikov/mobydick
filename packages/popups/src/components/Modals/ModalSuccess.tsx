import {rem, useTheme} from '@npm/mobydick-styles';
import React, {FC} from 'react';
import {IButtonTypes} from '@npm/mobydick-cta';

import {ModalBase} from '../ModalBase';
import {IContentProps, IHorizontalButtonsView} from '../../types';

interface IProps extends Partial<IHorizontalButtonsView> {
  title: string;
  descriptionText: string | null;
  buttonText?: string;
  onPressFirst?(): void;
  onPressSecond?(): void;
}

const ModalSuccess: FC<IContentProps & IProps> = props => {
  const {
    onClose,
    title,
    descriptionText,
    buttonText,
    typeLeft,
    onPressLeft,
    textLeft,
    typeRight,
    textRight,
    onPressRight,
    onPressFirst,
    onPressSecond,
  } = props;
  const {colors} = useTheme();

  return (
    <ModalBase overlayStyle={{justifyContent: 'center'}} {...props}>
      <ModalBase.AlertContent
        name={'icon-warning'}
        color={colors.IconAttention}
        style={{backgroundColor: colors.BgError}}
      />
      {title && <ModalBase.Title title={title} />}
      {descriptionText && (
        <ModalBase.DescriptionText descriptionText={descriptionText} />
      )}

      {buttonText && (
        <ModalBase.VerticalButtonsView>
          <ModalBase.VerticalButton
            type={IButtonTypes.tertiary}
            onPress={onClose}
            text={buttonText || 'Ok'}
          />
        </ModalBase.VerticalButtonsView>
      )}

      {textLeft && onPressLeft && textRight && onPressRight ? (
        <ModalBase.HorizontalButtonsView
          typeLeft={typeLeft || IButtonTypes.tertiary}
          textLeft={textLeft}
          onPressLeft={onPressLeft}
          typeRight={typeRight || IButtonTypes.destructive}
          textRight={textRight}
          onPressRight={onPressRight}
        />
      ) : null}

      <ModalBase.VerticalButtonsView>
        <ModalBase.VerticalButton
          onPress={onPressFirst}
          type={IButtonTypes.destructive}
          text={'Удалить'}
          style={{marginBottom: rem(12)}}
        />
        <ModalBase.VerticalButton
          onPress={onPressSecond}
          type={IButtonTypes.tertiary}
          text={'Отменить'}
        />
      </ModalBase.VerticalButtonsView>
    </ModalBase>
  );
};

export default ModalSuccess;
