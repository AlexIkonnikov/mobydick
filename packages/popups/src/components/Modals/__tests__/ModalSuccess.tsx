import {render} from '@testing-library/react-native';
import React from 'react';

import ModalSuccess from '../ModalSuccess';

describe('@npm/mobydick-popups/modalSuccess', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <ModalSuccess
        title={'title'}
        descriptionText={'descriptionText'}
        id={'id'}
        onClose={() => null}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with optional', () => {
    const {toJSON} = render(
      <ModalSuccess
        title={'title'}
        descriptionText={'descriptionText'}
        buttonText={'buttonText'}
        id={'id'}
        onClose={() => null}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
