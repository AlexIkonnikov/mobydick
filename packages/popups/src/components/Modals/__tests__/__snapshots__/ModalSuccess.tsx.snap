// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/modalSuccess should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      Array [
        Object {
          "justifyContent": "flex-end",
        },
        Object {
          "justifyContent": "center",
        },
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 18,
          "justifyContent": "center",
          "marginVertical": 36,
          "padding": 30,
          "width": 726,
        },
        Object {
          "maxWidth": 450,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(43, 120, 238, 0.15)",
            "borderRadius": 100,
            "padding": 18,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Object {
            "color": "#2B78EE",
            "fontFamily": "Neotis",
            "fontSize": 54,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 12,
          "marginTop": 18,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 27,
              "lineHeight": 33,
            },
            Array [
              Object {
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        title
      </Text>
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "#5E6678",
              "fontFamily": "Inter-Regular",
              "fontSize": 21,
              "lineHeight": 27,
            },
            Array [
              Object {
                "paddingTop": 12,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        descriptionText
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "marginTop": 30,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#2B78EE",
            "borderRadius": 18,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 72,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 27,
                "lineHeight": 33,
              },
              Array [
                Object {
                  "color": "#FFF",
                  "paddingHorizontal": 9,
                },
                undefined,
              ],
            ]
          }
        >
          OK
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-popups/modalSuccess should renders correctly with optional 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      Array [
        Object {
          "justifyContent": "flex-end",
        },
        Object {
          "justifyContent": "center",
        },
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 18,
          "justifyContent": "center",
          "marginVertical": 36,
          "padding": 30,
          "width": 726,
        },
        Object {
          "maxWidth": 450,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(43, 120, 238, 0.15)",
            "borderRadius": 100,
            "padding": 18,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Object {
            "color": "#2B78EE",
            "fontFamily": "Neotis",
            "fontSize": 54,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 12,
          "marginTop": 18,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 27,
              "lineHeight": 33,
            },
            Array [
              Object {
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        title
      </Text>
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "#5E6678",
              "fontFamily": "Inter-Regular",
              "fontSize": 21,
              "lineHeight": 27,
            },
            Array [
              Object {
                "paddingTop": 12,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        descriptionText
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "marginTop": 30,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#2B78EE",
            "borderRadius": 18,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 72,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 27,
                "lineHeight": 33,
              },
              Array [
                Object {
                  "color": "#FFF",
                  "paddingHorizontal": 9,
                },
                undefined,
              ],
            ]
          }
        >
          buttonText
        </Text>
      </View>
    </View>
  </View>
</View>
`;
